### PR TITLE
Exception code

### PR DESCRIPTION
**Pull request by roukmoute**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests?        | yes
| Doc?          | no
| Fixed tickets | https://github.com/algolia/algoliasearch-client-php/issues/283

### Description

This PR catches an Exception code to distinct different type of AlgoliaException..

### CODE REVIEW

**Comment by rayrutjes** [comment]
Hey, thanks @roukmoute ,
Can you add a test to cover that new code?

Also I'm unsure what the current error code is, and if users rely on that code already in which case this PR would introduce a BC somehow.

Happy to discuss it ;)

**Comment by roukmoute** [comment]
I don't think how it could introduce BC, because actually all code exceptions are at `0`.
So no one can use this code.

Now on the contrary it could help to know what kind of exception it is.

My [issue](#283) is a real concrete example that I have on a project.

**Comment by rayrutjes** [comment]
Thanks for the precisions @roukmoute , let's move forward with this then, I see no reason to not ge a proper exception code here.

I feel like there are other places where we could set the correct error codes like here: https://github.com/roukmoute/algoliasearch-client-php/blob/3f1a3b823e6d0d33373a807b393a5843aab626a1/src/AlgoliaSearch/Client.php#L992

If you can just add a unit test for the current code you have, then eventually address other error codes then I can take it over.

Best.

**Comment by roukmoute** [comment]
Here it is my update :)

**Comment by roukmoute** [comment]
ping @rayrutjes @maxiloc

**Comment by rayrutjes** [comment]
Thanks for the ping @roukmoute , we will take a look at it.
Please bear with us.

File: src/AlgoliaSearch/Client.php [hunk 1]
```diff
@@ -989,7 +989,7 @@ public function doRequest(
         curl_close($curlHandle);
 
         if (intval($http_status / 100) == 4) {
-            throw new AlgoliaException(isset($answer['message']) ? $answer['message'] : $http_status.' error');
+            throw new AlgoliaException(isset($answer['message']) ? $answer['message'] : $http_status.' error', $http_status);
         } elseif (intval($http_status / 100) != 2) {
             throw new \Exception($http_status.': '.$response);
```
Comment (JanPetr) [inline src/AlgoliaSearch/Client.php]: Shouldn't the code be added to this exception as well?

Comment (roukmoute) [inline src/AlgoliaSearch/Client.php]: Done @JanPetr

**Comment by roukmoute** [comment]
Done @JanPetr

**Comment by roukmoute** [comment]
Ping @rayrutjes @maxiloc @JanPetr

**Comment by rayrutjes** [comment]
Sorry about the delay, please check #298 .

**Comment by roukmoute** [comment]
Why have you closed my P.R.?

**Comment by rayrutjes** [comment]
Hey @roukmoute ,
Your commits have been pushed as part of #298, you will get credits there.

Because we are actually using secured tokens from Travis, we need to push code ourselves for the tests to run.

**Pull request closed without merging**